lld/ELF/Arch/RISCV.cpp

```diff
@@ -699,7 +699,7 @@ void RISCV::relocate(uint8_t *loc, const Relocation &rel, uint64_t val) const {
       warn("R_RISCV_CHERIOT_COMPARTMENT_HI relocation applied to instruction "
            "with unexpected opcode " +
            Twine(existingOpcode));
-    checkInt(loc, val, 20, rel);
+    checkInt(loc, SignExtend64(val + 0x800, bits) >> 12, 20, rel);
```

Comment:
Can this be `SignExtend64(hi20(val))`?

Reply:
Possibly. This is the same check as for AUIPCC and I wanted to be consistent.